### PR TITLE
Use URI library to parse hash and type MSKTabs strongly

### PR DESCRIPTION
**Pull request by alisman**
# What? Why?
Fix # .

Changes proposed in this pull request:
- a
- b

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: src/shared/lib/ExtendedRouterStore.ts [hunk 1]
```diff
@@ -35,16 +32,13 @@ export default class ExtendedRouterStore extends (RouterStore as FunctionConstru
             }
         });
```
Comment (alisman) [inline src/shared/lib/ExtendedRouterStore.ts]: I attempted to use the URI library to handle construction of the hash, but couldn't find a simple way to do it.   Also, I realized that $.param handles the uri encoding of parameter values, as demonstrated in test.  So, at this point, the only thing we're handling ourselves is the building of the parameter map, which seems very generic to me.

File: src/shared/lib/ExtendedRouterStore.ts [hunk 2]
```diff
@@ -2,19 +2,16 @@ import { RouterStore } from 'mobx-react-router';
...
     @computed get query(): {[key: string] : string} {
-        // var route = new RT(stores.routing.location.pathname);
-        var queryString: { [key: string] : string } = {};
-        decodeURIComponent(this.location.search).replace(
-            new RegExp("([^?=&]+)(=([^&]*))?", "g"),
-            function($0, $1, $2, $3) { queryString[$1] = $3; return $0}
-        )
-        return queryString;
+
+        return (URI.parseQuery(this.location.search) as {[key: string] : string});
```
Comment (alisman) [inline src/shared/lib/ExtendedRouterStore.ts]: A note, we already have at least 2 other URL building/parsing libraries installed.  So, at some point we should consolidate.  I will make a card for that.

File: src/shared/lib/ExtendedRouterStore.ts [hunk 3]
```diff
@@ -35,16 +32,13 @@ export default class ExtendedRouterStore extends (RouterStore as FunctionConstru
...
         let params = $.param(newQuery);
 
-        // if we have a new path, adopt it
-        // otherwise use current path
-
-        newPath = newPath || this.location.pathname;
-
-        newPath = (newPath[0] === '/') ? newPath : '/'+ newPath;
+        // put a leading slash if there isn't one
+        path = (path[0] === '/') ? path : '/' + path;
```
Comment (adamabeshouse) [inline src/shared/lib/ExtendedRouterStore.ts]: this is the kind of stuff that leads to issues in edge cases but this is probably simple enough to be fine

File: src/shared/lib/ExtendedRouterStore.ts (hunk 2 above)
Comment (adamabeshouse) [inline src/shared/lib/ExtendedRouterStore.ts]: actually I was just playing around and it looks like the query is already parsed, I guess as part of RouterStore. But this.location.query is a map of parameter names to url decoded parameter values.

**Pull request merged**